### PR TITLE
Fixes #13042: static-libgcc is ignored by some build

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/13042

### CODE REVIEW

File: rudder-agent/SPECS/rudder-agent.spec [hunk 1]
```diff
@@ -284,6 +284,7 @@ Requires: pcre
 # -L adds the library path to the produced executables
 # -static-libgcc is added because gcc is not provided on aix
 %define build_ldflags -static-libgcc -Wl,-brtl -Wl,-L%{rudderdir}/lib
+%define build_cflags -static-libgcc
```
Comment (peckpeck) [inline rudder-agent/SPECS/rudder-agent.spec]: shouldn't it be defined in the %else too ?

File: rudder-agent/SPECS/rudder-agent.spec [hunk 2]
```diff
@@ -322,7 +323,7 @@ cd %{_sourcedir}
 cp /usr/lib64/libattr.a /usr/lib64/libattr.la /lib64 || cp /usr/lib/libattr.a /usr/lib/libattr.la /lib
 %endif
 
-make %{?_smp_mflags} BUILD_CFLAGS="${RPM_OPT_FLAGS}" BUILD_LDFLAGS="%{build_ldflags}" USE_SYSTEM_OPENSSL=%{use_system_openssl} USE_SYSTEM_LMDB=%{use_system_lmdb} USE_SYSTEM_PCRE=%{use_system_pcre} USE_SYSTEM_FUSION=%{use_system_fusion} USE_SYSTEM_PERL=%{use_system_perl} USE_HTTPS=%{use_https} USE_SYSTEM_ZLIB=%{use_system_zlib} USE_SYSTEM_CURL=%{use_system_curl} USE_SYSTEM_YAML=%{use_system_yaml} USE_SYSTEM_XML=%{use_system_xml} USE_PIE=%{use_pie} USE_ACL=%{use_acl}
+make %{?_smp_mflags} BUILD_CFLAGS="${RPM_OPT_FLAGS} %{build_cflags}" BUILD_LDFLAGS="%{build_ldflags}" USE_SYSTEM_OPENSSL=%{use_system_openssl} USE_SYSTEM_LMDB=%{use_system_lmdb} USE_SYSTEM_PCRE=%{use_system_pcre} USE_SYSTEM_FUSION=%{use_system_fusion} USE_SYSTEM_PERL=%{use_system_perl} USE_HTTPS=%{use_https} USE_SYSTEM_ZLIB=%{use_system_zlib} USE_SYSTEM_CURL=%{use_system_curl} USE_SYSTEM_YAML=%{use_system_yaml} USE_SYSTEM_XML=%{use_system_xml} USE_PIE=%{use_pie} USE_ACL=%{use_acl}
```
Comment (amousset) [inline rudder-agent/SPECS/rudder-agent.spec]: BUILD_CFLAGS=

File: rudder-agent/SPECS/rudder-agent.spec (hunk 1 above)
Comment (amousset) [reply rudder-agent/SPECS/rudder-agent.spec]: it should, with an empty value

Comment (amousset) [reply rudder-agent/SPECS/rudder-agent.spec]: `%{nil}`

File: rudder-agent/SPECS/rudder-agent.spec (hunk 2 above)
Comment (peckpeck) [reply rudder-agent/SPECS/rudder-agent.spec]: where ?

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SPECS/rudder-agent.spec (hunk 2 above)
Comment (amousset) [reply rudder-agent/SPECS/rudder-agent.spec]: Sorry, misread it.

Comment (amousset) [reply rudder-agent/SPECS/rudder-agent.spec]: sorry, misread it.

**Pull request merged**